rules/android_application/android_application_rule.bzl

```diff
@@ -450,14 +450,16 @@ def android_application_macro(_android_binary, **attrs):
     # In the future bundle_config will accept a build rule rather than a raw file.
     bundle_config_file = bundle_config_file or bundle_config
 
+    deps = [d for d in attrs.pop("deps", [])]
     for feature_module in feature_modules:
         if not feature_module.startswith("//") or ":" not in feature_module:
             _log.error("feature_modules expects fully qualified paths, i.e. //some/path:target")
         module_targets = get_feature_module_paths(feature_module)
-        attrs["deps"].append(str(module_targets.title_lib))
+        deps.append(str(module_targets.title_lib))
```

Comment:
Actually this needs to change to: 
```
    deps = attrs.pop("deps", [])
    for feature_module in feature_modules:
        if not feature_module.startswith("//") or ":" not in feature_module:
            _log.error("feature_modules expects fully qualified paths, i.e. //some/path:target")
        module_targets = get_feature_module_paths(feature_module)
        deps = deps + [str(module_targets.title_lib)]
```

otherwise it will break the ability to use `select` on the `deps` attribute like so:
```
/external/build_bazel_rules_android/rules/android_application/android_application_rule.bzl", line 549, column 24, in android_application_macro
		deps = [d for d in attrs.pop("deps", [])]
Error: type 'select' is not iterable
```

cc @ahumesky @ted-xie